src/zeek.bif

```diff
@@ -401,6 +401,70 @@ function current_event_time%(%): time
 	return zeek::make_intrusive<zeek::TimeVal>(zeek::event_mgr.CurrentEventTime());
 	%}
 
+## Register the expected Zeek type for event metadata.
+##
+## id: The event metadata identifier.
+##
+## t: A type expression or type alias.
+##
+## Returns: true if the registration was successful, false if *id* is
+##          registered with a different type already, or type is *any*
+##          or a composite type containing *any*.
+##
+## .. zeek:see:: EventMetadata::current EventMetadata::current_all
+function EventMetadata::register%(id: EventMetadata::ID, t: any%): bool
+       %{
+       const auto& tty = t->GetType();
+
+       if ( tty->Tag() != zeek::TYPE_TYPE )
+               {
+               zeek::emit_builtin_error("register_event_metadata() expects type");
+               return zeek::val_mgr->False();
+               }
+
+       EnumValPtr idvp = {zeek::NewRef{}, id->AsEnumVal()};
+
+       bool r = zeek::event_registry->RegisterMetadata(idvp, tty->AsTypeType()->GetType());
+       return zeek::val_mgr->Bool(r);
+       %}
+
+## Query the current event's metadata with identifier *id*.
+##
+## id: The metadata identifier, e.g. ``EventMetadata::NETWORK_TIMESTAMP``.
+##
+## Returns: A vector of values. The vector is empty if no metadata with
+##          the given identifier is attached to this event, otherwise a
+##          vector whose elements are of the type used during registration.
+##
+## .. zeek:see:: EventMetadata::register EventMetadata::current_all
+function EventMetadata::current%(id: EventMetadata::ID%): any_vec
+	%{
+	static const auto& vt = zeek::id::find_type<zeek::VectorType>("any_vec");
+	return zeek::event_mgr.CurrentEvent()->MetadataValues({zeek::NewRef{}, id->AsEnumVal()});
+	%}
+
+## Query all of the current event's metadata.
+##
+## Returns: A vector :zeek:see:`EventMetadata::Entry` elements holding all
+##          the metadata attached to this event.
+##
+## .. zeek:see:: EventMetadata::register EventMetadata::current
+function EventMetadata::current_all%(%): event_metadata_vec
+	%{
+	static const auto& vt = zeek::id::find_type<zeek::VectorType>("event_metadata_vec");
+
+	auto result = zeek::make_intrusive<zeek::VectorVal>(vt);
+	const auto& mdv = zeek::event_mgr.CurrentEvent()->Metadata();
+
+	if ( mdv ) {
+		for ( const auto& entry : *mdv )
+			result->Append(entry.BuildVal());
+
+	}
```

Comment:
It seems `CurrentEvent` could return a `nullptr`. Let's either `assert` it does not, or handle it.

We could also slightly clean up where we introduce `mdv`.

```suggestion
	auto* event = zeek::event_mgr.CurrentEvent();
	if ( ! event )
		return result;

	if ( const auto* mdv = event->Metadata() ) {
 		for ( const auto& entry : *mdv )
 			result->Append(entry.BuildVal());
 	}
```

Reply:
Ouch, thanks!
```
$ zeek -e 'print EventMetadata::current_all()'
Segmentation fault (core dumped)
```